frontend/public/redux.ts

```diff
@@ -53,41 +58,33 @@ const store = createStore(
   composeEnhancers(applyMiddleware(thunk)),
 );
 
-const addPluginListener = () => {
-  const reducerExtensions = pluginStore.getAllExtensions().filter(isReduxReducer);
-  const getReduxFlagsObject = () => {
-    const gatingFlags = getGatingFlagNames(reducerExtensions);
-    const featureState = store.getState()[featureReducerName];
-    return featureState ? _.pick(featureState.toObject(), gatingFlags) : null;
-  };
-
-  let flagsObject = getReduxFlagsObject();
-
-  store.subscribe(() => {
-    const currentFlagsObject = getReduxFlagsObject();
-
-    if (JSON.stringify(flagsObject) !== JSON.stringify(currentFlagsObject)) {
-      flagsObject = currentFlagsObject;
-
-      const pluginReducerExtensions = reducerExtensions.filter((e) =>
-        isExtensionInUse(e, flagsObject),
-      );
-
-      const pluginReducers: ReducersMapObject = pluginReducerExtensions.reduce((map, e) => {
-        map[e.properties.namespace] = e.properties.reducer;
-        return map;
-      }, {});
-
-      const nextReducers: ReducersMapObject<RootState> = _.isEmpty(pluginReducers)
-        ? baseReducers
-        : { plugins: combineReducers(pluginReducers), ...baseReducers };
-
-      store.replaceReducer(combineReducers<RootState>(nextReducers));
-    }
-  });
-};
-
-addPluginListener();
+const pluginReducers: ReducersMapObject = {};
+
+subscribeToExtensions<ResolvedReduxReducer>(
+  extensionDiffListener((added, removed) => {
+    removed.forEach(
+      (e) => pluginReducers[e.properties.scope] && delete pluginReducers[e.properties.scope],
+    );
+
+    resolveExtensionsArray(added)
+      .then((exts) => {
+        exts.forEach(({ properties: { scope, reducer } }) => {
+          pluginReducers[scope] = reducer;
+        });
+
+        const nextReducers: ReducersMapObject<RootState> = _.isEmpty(pluginReducers)
+          ? baseReducers
+          : { plugins: combineReducers(pluginReducers), ...baseReducers };
+
+        store.replaceReducer(combineReducers<RootState>(nextReducers));
+      })
+      .catch((err) => {
```

Comment:
If I read this correctly, due to `Promise.all` behavior, if one of the extensions fails to get resolved but others get resolved OK, the other ones would still have no effect.

I'd suggest removing `resolveExtensions(Array)` helper entirely and simply use `resolveExtension` here directly.